crates/red_knot_python_semantic/src/types.rs

```diff
@@ -1438,6 +1441,18 @@ impl<'db> ClassType<'db> {
             })
     }
 
+    pub fn is_subclass_of(self, db: &'db dyn Db, other: ClassType) -> bool {
+        // TODO: we need to iterate over the *MRO* here, not the bases
+        (other == self)
+            || self.bases(db).any(|base| match base {
+                Type::Class(base_class) => base_class == other,
+                // `is_subclass_of` is checking the subtype relation, in which gradual types do not
+                // participate, so we should not return `True` if we find `Any/Unknown` in the
+                // bases.
```

Comment:
This means that if we want binary operations to actually consider the possibility that an `Any` inheriting type might be a subtype when found on the RHS, we'll have to do something a bit more complicated there than just `is_subclass_of`; there might need to be another `may_be_subclass_of` method that does respect `Any/Unknown`?

To be honest, though, I feel like the behavior of binary operations when the RHS inherits `Any` is an edge case that may never practically matter to anyone, so I don't think we should spend more time on it now.

Reply:
> To be honest, though, I feel like the behavior of binary operations when the RHS inherits `Any` is an edge case that may never practically matter to anyone, so I don't think we should spend more time on it now.

Agreed, I'm definitely fine with what we have for now, the TODO comments are pretty clear!